packages/@sanity/base/src/components/rovingFocus/useRovingFocus.ts

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import {useCallback, useEffect, useState} from 'react'
+import {RovingFocusProps} from '.'
+
+const EMPTY_ARRAY = []
+
+const MUTATION_ATTRIBUTE_FILTER = ['aria-hidden', 'disabled', 'href']
+
+const FOCUSABLE =
+  'a[href], button, input, textarea, select, details, [tabindex]:not([tabindex="-1"])'
+
+function getFocusableElements(element: HTMLElement) {
+  return [...(element.querySelectorAll(FOCUSABLE) as any)].filter(
+    (el) => !el.hasAttribute('disabled') && !el.getAttribute('aria-hidden="true"')
+  ) as HTMLElement[]
+}
+
+export function useRovingFocus(props: RovingFocusProps, dependencies?: unknown[]): undefined {
```

Comment:
can you add a JSDoc comment here explaining what this is with links back to the roving focus definition? i think it would help give future readers more context